src/apps/backend/modules/otp/types.ts

```diff
@@ -0,0 +1,61 @@
+import { PhoneNumber } from '../account/types';
+import { ApplicationError } from '../application';
+import { HttpStatusCodes } from '../http';
+
+export enum OtpStatus {
+  EXPIRED = 'EXPIRED',
+  PENDING = 'PENDING',
+  SUCCESS = 'SUCCESS',
+}
+
+export class Otp {
+  id: string;
+  otpCode: string;
+  phoneNumber: PhoneNumber;
+  status: OtpStatus;
+}
+
+export enum OtpErrorCode {
+  INCORRECT_OTP = 'OTP_ERR_01',
+  OTP_EXPIRED = 'OTP_ERR_02',
+  FAILED_TO_CREATE_OTP = 'OTP_ERR_03',
+}
+
+export type CreateOTPParams = {
+  phoneNumber: PhoneNumber;
+};
+
+export type VerifyOTPParams = {
+  otpCode: string;
+  phoneNumber: PhoneNumber;
+};
+
+export class OtpIncorrectError extends ApplicationError {
+  code: OtpErrorCode;
+
+  constructor() {
+    super('Incorrect OTP.');
```

Comment:
This is poor copyright message. Please use generative AI, Google, online research and other tools at your disposal to ensure this seems user friendly message. One of the principal is never to blame the user.

Kindly re-review your PR in detail @dev-rathore and ensure all copyright messages are done right.